### PR TITLE
GraphQl-309: [Checkout] Checkout Agreements

### PR DESCRIPTION
**Pull request by VitaliyBoyko**
### Description (*)
Added coverage for checkout agreements.

#309 Checkout Agreement coverage

Sample query: 
```
{
  checkoutAgreements {
    agreement_id
    name
    content
    content_height
    checkbox_text
    is_html
  }
}
```

### CODE REVIEW

**Comment by naydav** [comment]
@galaoleksandr Pls, test this PR
Thanks

File: app/code/Magento/CheckoutAgreementsGraphQl/etc/module.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Module/etc/module.xsd">
+    <module name="Magento_CheckoutAgreementsGraphQl">
+        <sequence>
+            <module name="Magento_GraphQl"/>
```
Comment (naydav) [inline app/code/Magento/CheckoutAgreementsGraphQl/etc/module.xml]: This sequence dependency is redundant

File: app/code/Magento/CheckoutAgreementsGraphQl/etc/schema.graphqls [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Copyright © Magento, Inc. All rights reserved.
+# See COPYING.txt for license details.
+
+type Query {
+    checkoutAgreements: [CheckoutAgreement] @resolver(class: "Magento\\CheckoutAgreementsGraphQl\\Model\\Resolver\\CheckoutAgreements") @doc(description: "The Checkout Agreements query returns information about a Checkout Agreements")
```
Comment (naydav) [inline app/code/Magento/CheckoutAgreementsGraphQl/etc/schema.graphqls]: `The Checkout Agreements query returns information about a Checkout Agreements`  :)

File: dev/tests/api-functional/testsuite/Magento/GraphQl/CheckoutAgreements/Api/CheckoutAgreementsListTest.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    checkbox_text
+    is_html
+  }
+}
+QUERY;
+
+        $response = $this->graphQlQuery($query);
+        $this->assertArrayHasKey('checkoutAgreements', $response);
+        $agreements = $response['checkoutAgreements'];
+        $this->assertEquals(1, count($agreements));
```
Comment (naydav) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/CheckoutAgreements/Api/CheckoutAgreementsListTest.php]: `assertCount`

File: app/code/Magento/CheckoutAgreementsGraphQl/Model/Resolver/CheckoutAgreements.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace Magento\CheckoutAgreementsGraphQl\Model\Resolver;
+
+use Magento\CheckoutAgreementsGraphQl\Model\Resolver\DataProvider\CheckoutAgreements as CheckoutAgreementsDataProvider;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+
+/**
+ * CMS page field resolver, used for GraphQL request processing
```
Comment (naydav) [inline app/code/Magento/CheckoutAgreementsGraphQl/Model/Resolver/CheckoutAgreements.php]: Please, fix description

File: dev/tests/api-functional/testsuite/Magento/GraphQl/CheckoutAgreements/Api/CheckoutAgreementsListTest.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        $this->assertEquals('Checkout Agreement (active)', $agreements[0]['name']);
+        $this->assertEquals('Checkout agreement content: <b>HTML</b>', $agreements[0]['content']);
+        $this->assertEquals('200px', $agreements[0]['content_height']);
+        $this->assertEquals('Checkout agreement checkbox text.', $agreements[0]['checkbox_text']);
+        $this->assertEquals(true, $agreements[0]['is_html']);
+    }
```
Comment (naydav) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/CheckoutAgreements/Api/CheckoutAgreementsListTest.php]: Please, add one test case: `try to get when agreements are not set. The result should be empty array`
Thanks

File: app/code/Magento/CheckoutAgreementsGraphQl/Model/Resolver/DataProvider/CheckoutAgreements.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    /**
+     * Get All Active Checkout Agreements Data
+     *
+     * @return array
+     */
+    public function getData(): array
+    {
+        $this->searchCriteriaBuilder->addFilter(AgreementInterface::IS_ACTIVE, true);
```
Comment (naydav) [inline app/code/Magento/CheckoutAgreementsGraphQl/Model/Resolver/DataProvider/CheckoutAgreements.php]: Need to add support of `store` (with tests coverage)

Also, looks using the collection (which we can consider as query API)
```
$agreements = $this->agreementCollectionFactory->create();
$agreements->addStoreFilter($this->_storeManager->getStore()->getId()); // TODO: store should be get from query context
 $agreements->addFieldToFilter('is_active', 1);
```

File: app/code/Magento/CheckoutAgreementsGraphQl/Model/Resolver/CheckoutAgreements.php [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        $checkoutAgreementsData = $this->checkoutAgreementsDataProvider->getData();
```
Comment (naydav) [inline app/code/Magento/CheckoutAgreementsGraphQl/Model/Resolver/CheckoutAgreements.php]: Need to take into account `checkout/options/enable_agreements (Enable Terms and Conditions)` (should be covered with API-functional tests)
Pay attention option value could be per `store view`

**Review by naydav — CHANGES_REQUESTED** [review]

File: app/code/Magento/CheckoutAgreementsGraphQl/etc/schema.graphqls [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    checkoutAgreements: [CheckoutAgreement] @resolver(class: "Magento\\CheckoutAgreementsGraphQl\\Model\\Resolver\\CheckoutAgreements") @doc(description: "The Checkout Agreements information")
+}
+
+type CheckoutAgreement @doc(description: "Defines all Checkout Agreement information") {
+    agreement_id: Int @doc(description: "Checkout Agreement identifier")
+    name: String @doc(description: "Checkout Agreement name")
+    content: String @doc(description: "Checkout Agreement content")
+    content_height: String @doc(description: "Checkout Agreement content height")
+    checkbox_text: String @doc(description: "Checkout Agreement checkbox tex")
+    is_html: Boolean @doc(description: "Is Checkout Agreement content in HTML format")
```
Comment (pmclain) [inline app/code/Magento/CheckoutAgreementsGraphQl/etc/schema.graphqls]: @naydav @VitaliyBoyko  I think we need to include `mode` here as an ENUM. In the existing checkout flow mode determines if a user must accept the agreements by clicking a checkbox input.
Current mode options: https://github.com/magento/graphql-ce/blob/e60d3a2d8ff3964510ecaa0f7617fa9e482236c4/app/code/Magento/CheckoutAgreements/Model/AgreementModeOptions.php#L10-L12

Rendering Logic
Template: https://github.com/magento/graphql-ce/blob/46bbdd0e3d5796e427899b96c189888423cb55d4/app/code/Magento/CheckoutAgreements/view/frontend/web/template/checkout/checkout-agreements.html#L10-L35
Component: https://github.com/magento/graphql-ce/blob/3fc2f9e4c9e558fd73a7999ddec74cd8efe45d93/app/code/Magento/CheckoutAgreements/view/frontend/web/js/view/checkout-agreements.js#L33-L35

**Pull request merged**